interfaces.go

```diff
@@ -165,35 +165,23 @@ func NewRequest(r *http.Request) *Request {
 	}
 	url := fmt.Sprintf("%s://%s%s", protocol, r.Host, r.URL.Path)
 
+	sendDefaultPii := CurrentHub().Client() != nil && CurrentHub().Client().Options().SendDefaultPII
+
 	var cookies string
 	var env map[string]string
-	headers := map[string]string{}
-
-	if client := CurrentHub().Client(); client != nil {
-		if client.Options().SendDefaultPII {
-			// We read only the first Cookie header because of the specification:
-			// https://tools.ietf.org/html/rfc6265#section-5.4
-			// When the user agent generates an HTTP request, the user agent MUST NOT
-			// attach more than one Cookie header field.
-			cookies = r.Header.Get("Cookie")
-
-			for k, v := range r.Header {
-				headers[k] = strings.Join(v, ",")
-			}
-
-			if addr, port, err := net.SplitHostPort(r.RemoteAddr); err == nil {
-				env = map[string]string{"REMOTE_ADDR": addr, "REMOTE_PORT": port}
-			}
-		}
-	} else {
```

Comment:
Thanks for the contribution and raising this issue!

Couldn't we move this block inside the client check instead? IMO, this would be more readable than introducing a new function.

Reply:
Thanks for review.

> this would be more readable than introducing a new function

Right, moved it into if block.
8373d13221838d40aa3d23e841655fd07532049b